### DISCUSSION TITLE
Is there any way to query a foreign table from another schema ?

### DISCUSSION THREAD

**Discussion by Khanhan2k2** [supabase/supabase#12997] [Questions]
Is there any way to query a foreign table from another schema ?
in schema public i have user table
then in schema ternant i have table order which has user_id column references to id of table user in schema public
so there is any way to get user(*) from order with JS library.
Thank you so much !

**Comment by GaryAustin1** [comment]
The js client only works on one schema at a time.  You could potential have two clients setup, read your id and then use it in the other client.  You would need to make your 2nd schema available to the API.   And this approach is not using foreign keys, just fetching data with columns.

The better way is to use an rpc call to a postgres function which can use SQL with foreign keys in different schema.

**Reply by Khanhan2k2** [reply]
I'm glad to recieve reply from you, can i have one more question ?
How can i get list of schema name of my database
when i use this query:
SELECT schema_name
FROM information_schema.schemata;
In sql editor, it work perfectly
Then i create function for returning that, and call it with rpc with js library, i just receive public 
Please help me
Thank you so much

**Reply by GaryAustin1** [reply]
The API does not have permissions on any other schemas.  If you have custom schemas then you need to have grants set up on them to allow authenticated and/or anon roles to have access. Or you can use a “security definer” function to do the SQL you showed.   That runs as the owner so would have permission normally.

**Comment by steve-chavez** [comment]
Also see https://github.com/orgs/supabase/discussions/12997
